andela/ah-the-phoenix
# #163388135 Enable a user to reset their password via email

#### What does this PR do?
It enables a user to reset their password via email if at any point they forget it.
#### Description of Task to be completed?
- write tests for the password reset functionality
- create  a route and functionality for password resetting
- create mail sending functionality
- create a route and functionality for updating to the new set password
#### How should this be manually tested?
1. Clone the repo
```
git clone https://github.com/andela/ah-the-phoenix.git
```
2. Change it into the current directory
```
cd ah-the-phoenix
```
3. Checkout to the Feature branch
```    
git checkout ft-email-password-reset-163388135
```
4. Run the server
`python manage.py runserver`

5. Register a user on the following url
`http://127.0.0.1:8000/api/v1/users/`
      Sample data 
```
{
  "user": {
    "email": "savalijey@gmail.com",
    "password": "mypassword",
    "username": "James Savali"
  }
}
```
6. Send a password reset url to the mail through the following url
`http://127.0.0.1:8000/api/v1/users/password_reset/`
     Sample data to provide
```
{
   "email": "savalijey@gmail.com"
}
```

7. Check email for a reset password link

8. Click on the link and enter a new password and a confirmation password.
#### Any background context you want to provide?
None
#### What are the relevant pivotal tracker stories?
[#163388135 ](https://www.pivotaltracker.com/story/show/163388135)

#### Screenshots
<img width="1278" alt="screen shot 2019-02-07 at 11 50 33" src="https://user-images.githubusercontent.com/42576922/52400006-f0050080-2ace-11e9-9deb-9be18688d68d.png">
<img width="960" alt="screen shot 2019-02-07 at 11 51 47" src="https://user-images.githubusercontent.com/42576922/52400017-f98e6880-2ace-11e9-9608-a05fd7758ed0.png">
<img width="1280" alt="screen shot 2019-02-07 at 11 52 32" src="https://user-images.githubusercontent.com/42576922/52400023-00b57680-2acf-11e9-907b-f46f263fa28f.png">
